pkg/clusteragent/admission/patch/patcher.go

```diff
@@ -137,6 +137,10 @@ func disableConfig(deploy *corev1.Deployment, req PatchRequest) {
 		log.Debugf("Found pod label %q=%q in target %s. Setting it to false", common.EnabledLabelKey, val, req.K8sTarget)
 	}
 	deploy.Spec.Template.Labels[common.EnabledLabelKey] = "false"
+	if deploy.Spec.Template.Annotations == nil {
+		return
```

Comment:
I think it is not a return that need to be done here, but create the map, like it is done for labels map line 134

Reply:
Done in the last commit!